rclcpp/test/rclcpp/test_parameter.cpp

```diff
@@ -67,6 +70,12 @@ TEST_F(TestParameter, not_set_variant) {
   EXPECT_THROW(
     not_set_variant.get_value<int>(),
     rclcpp::exceptions::InvalidParameterTypeException);
+
+  EXPECT_EQ(not_set_variant, not_set_variant);
```

Comment:
This comparison to itself looks odd, I suppose you are testing a == operator? Please add a comment or maybe remove this in favor of the test checking `not_set_variant == not_set_variant2`

Reply:
I won't block PR on this though

Reply:
Yah, this is checking that `==` does in fact agree that the same object is equal to itself.

Reply:
I added a comment though, which is a great idea.